src/monodroid/jni/inmemory-assemblies.cc

```diff
@@ -0,0 +1,154 @@
+#include "inmemory-assemblies.hh"
+#include "globals.hh"
+
+#include <mono/metadata/appdomain.h>
+#include <mono/metadata/assembly.h>
+#include <mono/metadata/reflection.h>
+
+extern "C" {
+#include "java-interop-util.h"
+}
+
+namespace xamarin::android::internal {
+
+void
+InMemoryAssemblies::add_or_update_from_java (MonoDomain *domain, JNIEnv *env, jstring_array_wrapper &assemblies, jobjectArray assembliesBytes)
+{
+	if (assembliesBytes == nullptr)
+		return;
+
+	unsigned int len = static_cast<unsigned int> (env->GetArrayLength (assembliesBytes));
```

Comment:
why `static_cast<unsigned int>` instead of just using `jsize`?

Reply:
This is to prevent a cascading amount of warnings due to signed -> unsigned conversions (that was likely enabled by the recent changes in warning flags)